app/models/host.rb

```diff
@@ -1939,17 +1939,15 @@ def self.node_types
   end
 
   def self.openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    ems.empty? ? false : Host.where(:ems_id => ems).exists?
+    ManageIQ::Providers::Openstack::InfraManager::Host.count > 0
   end
 
   def self.non_openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where.not(:ems_id => ems).exists?
+    (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0
```

Comment:
I think (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0 is the one that is more correct.

Background: the code was there because there was requirement to rename Host to Node and Cluster to Deployment role, when you have only ManageIQ::Providers::Openstack::InfraManager in you ManageIQ. But it should go back to host, cluster when you have multiple providers.

So I would prefer (EmsInfra.count - ManageIQ::Providers::Openstack::InfraManager.count) > 0, so the naming doesn't change according to added hosts, but rather according to what providers are added. (if we still actually need this renaming :-) )